File: Studio.UI/src/app/data/results/events/EventResult.ts
import { Guid } from "guid-typescript";
import { WeddingResult } from "../weddings/WeddingResult";
import { LocationResult } from "../locations/LocationResult";
import { PhotoResult } from "../photos/PhotoResult";
import { ServiceResult } from "../services/ServiceResult";
import { BaseResult } from "../base/BaseResult";

export interface EventResult extends BaseResult{
    eventTitle: string;
    eventDescription?: string | null;
    weddingId?: Guid | null;
    locationId?: Guid | null;
    status?: string | null;
    wedding?: WeddingResult | null;
    location?: LocationResult | null;
    photos: PhotoResult[];
    services: ServiceResult[];
}

File: Studio-BE/src/NM.Studio.Domain/Results/Events/EventResult.cs
using NM.Studio.Domain.Entities.Weddings;
using NM.Studio.Domain.Results.Bases;
using NM.Studio.Domain.Results.Locations;
using NM.Studio.Domain.Results.Photos;
using NM.Studio.Domain.Results.Services;
using NM.Studio.Domain.Results.Weddings;

namespace NM.Studio.Domain.Results.Events
{
    public class EventResult : BaseResult
    {
        public string EventTittle { get; set; }
        
        public string? EventDescription { get; set; } 
        
        public Guid? WeddingId { get; set; }
        
        public Guid? LocationId { get; set; }
        
        public string? Status { get; set; }

        public WeddingResult Wedding { get; set; }
        
        public LocationResult Location { get; set; }

        public IList<EventXServiceResult> EventXServices { get; set; }

        public IList<EventXPhotoResult> EventXPhotos { get; set; }
    }
}
